framework/src/base/MooseApp.C

```diff
@@ -775,6 +789,16 @@ MooseApp::MooseApp(const InputParameters & parameters)
   Moose::out << std::flush;
 }
 
+std::optional<MooseEnum>
+MooseApp::getComputeDevice()
+{
+  if (isParamSetByUser("libtorch_device"))
+    return getParam<MooseEnum>("libtorch_device");
+  if (isParamSetByUser("compute_device"))
+    return getParam<MooseEnum>("compute_device");
+  return {};
```

Comment:
Why do we benefit from having this optional? Don't we want to just return `compute_device` as it has a default?